src/toil/lib/pipes.py

```diff
@@ -15,64 +15,64 @@
 
 class WritablePipe(ABC):
     """
-    An object-oriented wrapper for os.pipe. Clients should subclass it, implement
-    :meth:`.readFrom` to consume the readable end of the pipe, then instantiate the class as a
-    context manager to get the writable end. See the example below.
-
-    >>> import sys, shutil
-    >>> class MyPipe(WritablePipe):
-    ...     def readFrom(self, readable):
-    ...         shutil.copyfileobj(codecs.getreader('utf-8')(readable), sys.stdout)
-    >>> with MyPipe() as writable:
-    ...     _ = writable.write('Hello, world!\\n'.encode('utf-8'))
-    Hello, world!
-
-    Each instance of this class creates a thread and invokes the readFrom method in that thread.
-    The thread will be join()ed upon normal exit from the context manager, i.e. the body of the
-    `with` statement. If an exception occurs, the thread will not be joined but a well-behaved
-    :meth:`.readFrom` implementation will terminate shortly thereafter due to the pipe having
-    been closed.
-
-    Now, exceptions in the reader thread will be reraised in the main thread:
-
-    >>> class MyPipe(WritablePipe):
-    ...     def readFrom(self, readable):
-    ...         raise RuntimeError('Hello, world!')
-    >>> with MyPipe() as writable:
-    ...     pass
-    Traceback (most recent call last):
-    ...
-    RuntimeError: Hello, world!
-
-    More complicated, less illustrative tests:
-
-    Same as above, but proving that handles are closed:
-
-    >>> x = os.dup(0); os.close(x)
-    >>> class MyPipe(WritablePipe):
-    ...     def readFrom(self, readable):
-    ...         raise RuntimeError('Hello, world!')
-    >>> with MyPipe() as writable:
-    ...     pass
-    Traceback (most recent call last):
-    ...
-    RuntimeError: Hello, world!
-    >>> y = os.dup(0); os.close(y); x == y
-    True
-
-    Exceptions in the body of the with statement aren't masked, and handles are closed:
-
-    >>> x = os.dup(0); os.close(x)
-    >>> class MyPipe(WritablePipe):
-    ...     def readFrom(self, readable):
-    ...         pass
-    >>> with MyPipe() as writable:
-    ...     raise RuntimeError('Hello, world!')
-    Traceback (most recent call last):
-    ...
-    RuntimeError: Hello, world!
-    >>> y = os.dup(0); os.close(y); x == y
-    True
+    # An object-oriented wrapper for os.pipe. Clients should subclass it, implement
```

Comment:
I don't think it makes sense to try to have a big comment *inside* a docstring like this.